src/MDI/mdi_engine.cpp

```diff
@@ -18,322 +18,1662 @@
 
 #include "mdi_engine.h"
 
+#include <limits>
```

Comment:
@sjplimp these changes undo updates to upstream that had aligned this code with the contributing guidelines about include file names and ordering.